### ISSUE TITLE
Telegram webhook error when commit message contains special characters

### ISSUE THREAD

**Issue by babykart** [go-gitea/gitea#31182]
### Description

Telegram webhook respond an error code 400 when commit message contains special characters.
Example message:

```json
{
  "text": "[\u003ca href=\"https://git.example.com/babykart/et0ie\"\u003eKubler/et0ie\u003c/a\u003e:\u003ca href=\"https://git.example.com/babykart/et0ie/commit/6e873e503dd22aa4bf5e577bc9aa348e7b842cf7\"\u003emain\u003c/a\u003e] 1 new commit\n[\u003ca href=\"https://git.example.com/babykart/et0ie/commit/6e873e503dd22aa4bf5e577bc9aa348e7b842cf7\"\u003e6e873e5\u003c/a\u003e] [et0ie] caddy-24.05.301\n\nSigned-off-by: babykart \u003cbabykart@gmail.com\u003e - babykart",
  "parse_mode": "HTML",
  "disable_web_page_preview": true
}
```
Response:
```json
{"ok":false,"error_code":400,"description":"Bad Request: can't parse entities: Unsupported start tag \"babykart@gmail.com\" at byte offset 326"}
```
These messages perfectly work with the version 1.21.11.

### Gitea Version

1.22.0

### Can you reproduce the bug on the Gitea demo site?

Yes

### Git Version

2.45.1

### Operating System

Gentoo Linux

### How are you running Gitea?

Downloaded from https://github.com/go-gitea/gitea/releases/download/v1.22.0/gitea-1.22.0-linux-amd64

### Database

PostgreSQL

**Comment by POPSuL** [comment]
https://github.com/go-gitea/gitea/pull/29864
That fix broke it.

**Comment by wxiaoguang** [comment]
Hmm, a PR approved by me .... so let's fix it :  Use old behavior for telegram webhook #31588

**Comment by wxiaoguang** [comment]
" Use old behavior for telegram webhook #31588 " has been merged, when the next 1.22-nightly build is ready, please take a try.
